packages/client/src/ipfs/index.ts

```diff
@@ -3,34 +3,85 @@ import { IPFSClientConfig } from '../types';
 export default class IPFSClient {
   static readonly IPFS_CLIENT_ERROR = 'IPFS client not initialised';
   ipfs: any;
-  authorization: any;
+  authorization: string | undefined; 
+  provider: string | undefined;
+  initialized: boolean = false;
 
   constructor(ipfsClientConfig: IPFSClientConfig) {
-    const authorization =
-      'Basic ' + btoa(ipfsClientConfig.clientId + ':' + ipfsClientConfig.clientSecret);
-    this.authorization = authorization;
-    // ipfs-http-client is being mocked by src/__mocks__/ipfs-http-client
-    // due to the way its being imported in ipfs/index.ts, we have to add a special mock for it
-    import('ipfs-http-client').then(({ create }) => {
-      this.ipfs = create({
-        url: ipfsClientConfig.baseUrl,
-        headers: {
-          authorization,
-        },
+    this.provider = ipfsClientConfig.provider;
+    if (ipfsClientConfig.provider === 'quicknode') {
+      this.authorization = ipfsClientConfig.apiKey;
+      this.initialized = true;
+    }
+    else if (ipfsClientConfig.provider === 'infura') {
+      const authorization =
+        'Basic ' + btoa(ipfsClientConfig.clientId + ':' + ipfsClientConfig.clientSecret);
+      this.authorization = authorization;
+      import('kubo-rpc-client').then(({ create }) => {
```

Comment:
can you check if this dynamic import can be converted to a static one?

we used to import ipfs-http-client this way because it had some issues. if kubo does not have those issues, we can import it on the top of the file like a regular import